### PR TITLE
Deploy Files to CDN

### PR DESCRIPTION
**Pull request by bigtimebuddy**
## Overview

This change uses Travis and S3 to automatically deploy all PRs and Commits to the a Pixi S3 bucket. It will allow us to remove bin files from the commits (no more mistaken PRs with bin files). Users could now pull ANY commit into a fiddle/jsbin. Should _really_ improve testing. 

Here's an example of a deploy script:
https://s3-eu-west-1.amazonaws.com/pixi-build/3d7c1d2956214351dd4b3d3da4d7eb71fc1b6d0d/pixi.min.js

This maps to this push:
https://github.com/pixijs/pixi.js/commit/3d7c1d2956214351dd4b3d3da4d7eb71fc1b6d0d

### CODE REVIEW

**Comment by bigtimebuddy** [comment]
cc @GoodBoyDigital & @ivanpopelyshev

**Comment by ivanpopelyshev** [comment]
Can we also reference them from pixi-examples somehow?

**Comment by bigtimebuddy** [comment]
Whenever we push a tag, it will also generate a release for it that we should be able to reference by tag name: for instance:

https://s3-eu-west-1.amazonaws.com/pixi-release/v4.0.0/pixi.min.js (this doesn't work yet).

I think we can use these tag release in the pixi-examples.

**Comment by bigtimebuddy** [comment]
Another option, that's be easy to add would be an easy way to fetch the latest build from a specific branch:
https://s3-eu-west-1.amazonaws.com/pixi-build/dev/pixi.min.js

**Comment by ivanpopelyshev** [comment]
yep, that's what i'm talking about. Nice. Now I have to think how to add plugins there

**Comment by bigtimebuddy** [comment]
Okay, I changed the bucket names slightly to distinguish between branch/commit/tag.

* https://s3-eu-west-1.amazonaws.com/pixi.js-branch/[branch]/pixi.min.js
* https://s3-eu-west-1.amazonaws.com/pixi.js-commit/[commit]/pixi.min.js
* https://s3-eu-west-1.amazonaws.com/pixi.js-tag/[tag]/pixi.min.js

I think we should create a DNS mapping for pixijs.com that maps to these URLs:

* https://cdn.pixijs.com/branch/[branch]/pixi.min.js
* https://cdn.pixijs.com/commit/[branch]/pixi.min.js
* https://cdn.pixijs.com/tag/[branch]/pixi.min.js

**Comment by ivanpopelyshev** [comment]
Also we have to reference latest binary somewhere, so people wont ask "where did 'bin' folder go"

**Comment by bigtimebuddy** [comment]
@ivanpopelyshev Yah, getting the branch release would give you the latest bin.

File: .travis.yml [hunk 1]
```diff
@@ -25,4 +25,17 @@ before_script:
...
+    access_key_id: $S3_ACCESS_KEY_ID
+    secret_access_key: $S3_SECRET_ACCESS_KEY
+    bucket: "pixi.js"
+    skip_cleanup: true
+    acl: public_read
+    local_dir: bin
+    upload-dir: "$TRAVIS_BRANCH"
+    region: eu-west-1
+    on:
+        all_branches: true
```
Comment (englercj) [inline .travis.yml]: Do you need an `on` condition at all if you are just doing all branches?

Comment (bigtimebuddy) [inline .travis.yml]: I don't know, I found the docs for Travis to be not super clear on this point.

**Comment by bigtimebuddy** [comment]
Just FYI, @GoodBoyDigital and I decided to delay the ignoring of the **bin** folder to make the transition easier for developers. Once we start promoting and people start using the Travis-build files, and find a good solution for Bower, we can remove the **bin** folder.

**Pull request merged**